### PR TITLE
Connection Manager

### PR DESCRIPTION
**Pull request by cuminandpaprika**

### CODE REVIEW

File: ConnectionManager.lib.nut [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
 
 class ConnectionManager {
-    static version = [1,1,0];
+    static VERSION = "1.1.0";
```
Comment (ppetrosh) [inline ConnectionManager.lib.nut]: Strictly speaking, this is a breaking change so please update the version to 2.0.0.

File: ConnectionManager.lib.nut [hunk 2]
```diff
@@ -1,15 +1,36 @@
+// The MIT License (MIT)
```
Comment (ppetrosh) [inline ConnectionManager.lib.nut]: Can you please use the license file and headers from https://electricimp.com/docs/libraries/submissions/?

Comment (ppetrosh) [inline ConnectionManager.lib.nut]: All the files: [LICENSE](./LICENSE), [Example](https://github.com/electricimp/ConnectionManager/blob/master/example/ConnectionManager.example.nut)  and the the library file need to be updated.

File: ConnectionManager.lib.nut [hunk 3]
```diff
@@ -49,13 +69,19 @@ class ConnectionManager {
...
+        	case START_NO_ACTION:
+	        	// Do nothing
+	        	break;
+        	case START_CONNECTED:
+	        	// Start connecting if they ask for it
+	            imp.wakeup(0, connect.bindenv(this));
+	            break;
+        	case START_DISCONNECTED:
+	        	// Disconnect if required
+	            server.disconnect();
```
Comment (ppetrosh) [inline ConnectionManager.lib.nut]: I believe a) it should be `disconnect()` as it takes care of flushing the cache and some other stuff before disconnect b) it should be called the same way as connect is called above.

File: ConnectionManager.lib.nut [hunk 4]
```diff
@@ -79,7 +82,7 @@ class ConnectionManager {
 	            break;
         	case START_DISCONNECTED:
```
Comment (ppetrosh) [inline ConnectionManager.lib.nut]: There is something wrong with indents in this switch block. Could you please fix it?

File: ConnectionManager.lib.nut [hunk 5]
```diff
@@ -79,7 +82,7 @@ class ConnectionManager {
...
 	        	// Disconnect if required
-	            server.disconnect();
+                imp.wakeup(0, disconnect.bindenv(this));
 	            _connected = false;
```
Comment (ppetrosh) [inline ConnectionManager.lib.nut]: I don't think you need to clean up the `_connected` flag here.

File: ConnectionManager.lib.nut (hunk 4 above)
Comment (cuminandpaprika) [inline ConnectionManager.lib.nut]: Yup, fixed it.

File: ConnectionManager.lib.nut (hunk 5 above)
Comment (cuminandpaprika) [inline ConnectionManager.lib.nut]: Yup, removed it.

**Pull request merged**